### ISSUE TITLE
Exception in WAL playback: Catalog Error: Table does not exist

### ISSUE THREAD

**Issue by dbeatty10** [jwills/dbt-duckdb#17]
Hey @jwills

I got the following error when [hacking on the demo](https://github.com/dbeatty10/jaffle_shop_with_fal/tree/dbeatty/reproduce-bug) published by the fal.ai folks and using the dbt-duckdb 1.2.1 multi-threading:

```
Exception in WAL playback: Catalog Error: Table with name customers__dbt_tmp does not exist!
Did you mean "customers"?
```

This demo is using a forked and customized beta version of dbt-core + the demo fal adapter + dbt-duckdb multi-threading, so who knows what is _actually_ causing the issues -- may or may not be fixable within dbt-duckdb 🤷 

### Instructions to reproduce
Grab the code:
```
git clone https://github.com/dbeatty10/jaffle_shop_with_fal.git
cd jaffle_shop_with_fal
git checkout dbeatty/reproduce-bug
```

Create a virtual environment:
```shell
python3 -m venv env
source env/bin/activate
pip install --upgrade pip
```

Install the requirements and build:
```
pip install -r requirements.txt
source env/bin/activate
rm jaffle_shop.duckdb
dbt build
```

Deleting the `jaffle_shop.duckdb` file was crucial to reproduce the error for me -- if all the output tables exist already, everything works fine. 

### Work-around
Editing my `requirements.txt` to this and re-running the install+build steps worked again:
```
# Works with 1.2.0
dbt-duckdb==1.2.0

# Doesn't work with 1.2.1
# dbt-duckdb==1.2.1

# Use a local editable version of dbt-duckdb instead:
# -e /Users/dbeatty/projects/dbt-duckdb
```

**Comment by jwills** [comment]
Hey @dbeatty10 thanks again for finding this; I worked with the fal folks and figured out what was going on. Do you mind pulling in that change I made in #18 and confirm that it fixes the problem for you?

**Comment by dbeatty10** [comment]
I'll pull in #18 and confirm it works.

**Comment by dbeatty10** [comment]
@jwills thanks for the fix!

#18 worked for me as a solution 👍 

✅  Confirmed that my [bug demo branch](https://github.com/dbeatty10/jaffle_shop_with_fal/tree/dbeatty/reproduce-bug) was fixed via #18 by changing `requirements.txt` to use the following:
```
dbt-duckdb @ git+https://github.com/jwills/dbt-duckdb.git@jwills_change_this_up_a_bit
```
